### ISSUE TITLE
Regarding Loss Function using ground truth labels in linear_loss part

### ISSUE THREAD

**Issue by SonalKumar95** [mhamilton723/STEGO#89]
Hello there,

Please let me know if I'm wrong. In the train_segmentation.py, the loss function includes two extra loss functions, i.e., linear_loss and cluster_loss. For unsupervised training, the cluster_loss seems okay, but the linear_loss uses the ground truth labels. Is it the case or I'm getting it wrong? Please help me.

Thanks in advance.

**Comment by mhamilton723** [comment]
Hey Sonal, thats okay because theres a detach and clone before the linear
loss. Its basically just an entirely separate probe component whose
gradients dont alter the training of the unsupervised components.

**Comment by bio-mlhui** [comment]
I am also confused. The `detached_code = torch.clone(model_output[1].detach())` will generate `detached_code` which does not requires grad. However, the `linear_output = self.linear_model(detached_code)` will generate `linear_output` which requires grad. Since `linear_output` is used to compute `linear_loss` which uses ground truth mask labels, the final loss will backward its gradient to the `linear_model`. Does this mean the final algorithm is not unsupervised?
